src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/LoggerTests.cs

```diff
@@ -0,0 +1,93 @@
+            clockMock = new Mock<IClock>();
+            logger = new Logger(clockMock.Object);
+        }
+
+        [Fact]
+        public void TestLogInfo()
+        {
+            using var consoleOutput = new StringWriter();
+            var message = "some message";
+            var time = DateTimeOffset.MinValue;
+            clockMock.SetupGet(obj => obj.Now).Returns(time);
+            var expectedLog = $"{time} [INFO]: some message" + Environment.NewLine;
+            Console.SetOut(consoleOutput);
+            logger.LogInfo(message);
+            var actualLog = consoleOutput.ToString();
+            Assert.Equal(expectedLog, actualLog);
+        }
+
+        [Fact]
+        public void TestLogError()
+        {
+            using var consoleOutput = new StringWriter();
+            var message = "some message";
+            var time = DateTimeOffset.MinValue;
+            clockMock.SetupGet(obj => obj.Now).Returns(time);
+            var expectedLog = $"{time} [ERROR]: some message" + Environment.NewLine;
+            Console.SetOut(consoleOutput);
+            logger.LogError(message);
+            var actualLog = consoleOutput.ToString();
+            Assert.Equal(expectedLog, actualLog);
+        }
+
+        [Fact]
+        public void TestLogExceptionWithoutStackTrace()
+        {
+            using var consoleOutput = new StringWriter();
+            var time = DateTimeOffset.MinValue;
+            clockMock.SetupGet(obj => obj.Now).Returns(time);
+            var exception = new InvalidOperationException();
+            var expectedLog = $"{time} [ERROR]: " +
+                "Exception encountered: System.InvalidOperationException: " +
+                exception.Message + Environment.NewLine + exception.StackTrace + Environment.NewLine;
+            Console.SetOut(consoleOutput);
+            logger.LogException(exception);
+            var actualLog = consoleOutput.ToString();
+            Assert.Equal(expectedLog, actualLog);
+        }
+
+        [Fact]
+        public void TestLogExceptionWithStackTrace()
+        {
+            using var consoleOutput = new StringWriter();
+            var time = DateTimeOffset.MinValue;
+            clockMock.SetupGet(obj => obj.Now).Returns(time);
+            Exception exception;
+            try
+            {
+                throw new InvalidOperationException();
+            }
+            // Throw exception to generate stack trace.
+            catch (Exception thrownException)
+            {
+                exception = thrownException;
+            }
+
+            var expectedLog = $"{time} [ERROR]: " +
+                "Exception encountered: System.InvalidOperationException: " +
+                exception.Message + Environment.NewLine + exception.StackTrace + Environment.NewLine;
+            Console.SetOut(consoleOutput);
+            logger.LogException(exception);
+            var actualLog = consoleOutput.ToString();
+            Assert.Equal(expectedLog, actualLog);
```

Comment:
All tests in this file seems rather strict. It's like writing logger twice. Do we have log format documented? If not, does it make sense to make tests less restrictive? For example, make sure that certain elements appear somewhere in the output.

Reply:
Made the tests less strict by just making sure that the elements that should be in the log are in the log. No specific format is checked.